src/andromede/simulation/optimization.py

```diff
@@ -747,23 +758,46 @@ def _create_variables(self) -> None:
                                 instantiated_ub_expr
                             ).get_value(block_timestep, scenario)
 
-                        # TODO: Add BoolVar or IntVar if the variable is specified to be integer or bool
+                        scenario_suffix = (
+                            f"_s{scenario}"
+                            if var_indexing.is_scenario_varying()
+                            and (self.context.scenarios > 1)
+                            else ""
+                        )
+
+                        # Set solver var name
                         # Externally, for the Solver, this variable will have a full name
                         # Internally, it will be indexed by a structure that into account
                         # the component id, variable name, timestep and scenario separately
                         solver_var = None
-                        if model_var.data_type == ValueType.FLOAT:
-                            solver_var = self.solver.NumVar(
-                                lower_bound,
-                                upper_bound,
-                                f"{component.id}_{model_var.name}_t{block_timestep}_s{scenario}",
+                        solver_var_name = f"{component_prefix}{var_name}{block_suffix}{scenario_suffix}"
+
+                        if math.isclose(lower_bound, upper_bound):
+                            raise ValueError(
+                                f"Upper and lower bounds of variable {solver_var_name} have the same value: {lower_bound}"
```

Comment:
I think this could be an OK situation, where for example you have 2 timeseries that constrain your variable up and down, and for some timesteps the 2 bounds are equal (think rule curves of hydro reservoirs where you want to enforce a value for some timesteps).